sark-functions.sh

```diff
@@ -527,31 +527,37 @@ done
 }
 
 purge_old_packages() {
-export DOCKER_OPTS="-t --name ${REPOSITORY_NAME}-removeclean-${JOB_ID}"
-local PKGLISTS=($(find ${VAGRANT_DIR}/artifacts/$REPOSITORY_NAME/ | grep PKGLIST))
-local REMOVED=0
-for i in "${PKGLISTS[@]}"
-do
-  local REPO_CONTENT=$(cat ${i} | perl -lpe 's:\~.*::g' | xargs echo );
-  local TOREMOVE=$(OUTPUT_REMOVED=1 PACKAGES=$REPO_CONTENT sark-version-sanitizer );
-  [ -n "${TOREMOVE}" ] && let REMOVED+=1 && package_remove ${TOREMOVE}
-done
+  export DOCKER_OPTS="-t --name ${REPOSITORY_NAME}-removeclean-${JOB_ID}"
+  local PKGLISTS=($(find ${VAGRANT_DIR}/artifacts/$REPOSITORY_NAME/ | grep PKGLIST))
+  local REMOVED=0
+  for i in "${PKGLISTS[@]}"
+  do
+    local REPO_CONTENT=$(cat ${i} | perl -lpe 's:\~.*::g' | xargs echo );
+    local TOREMOVE=$(OUTPUT_REMOVED=1 PACKAGES=$REPO_CONTENT sark-version-sanitizer );
+    [ -n "${TOREMOVE}" ] && let REMOVED+=1 && package_remove ${TOREMOVE}
+  done
+
+  [ $REMOVED != 0 ] && generate_repository_metadata
+  [ "$DOCKER_COMMIT_IMAGE" = true ] && \
```

Comment:
If docker commit fails then docker rm gets executed, and it works against set -e.

Reply:
mmm... I think that *set -e* gonna be defined on father file that include sark-functions.sh. See for example sark-localbuild.

Reply:
I meant two things: first, `condition && cmd1 || cmd2` is not a direct replacement of `if then else`: "If docker commit fails then docker rm gets executed."

Second, if docker commit fails you probably want the script to exit according to set -e, but here instead of that, docker rm will be executed.

In this case the script will probably fail at some other point but it's good to catch errors as soon as they appear. Tl;dr use if/else.

Reply:
For that section I only fix indentation :P It's not currently used by sark-localbuild and for imho we can remove it after complete migration to mottainai :)

Personally, I don't like maintains code never used.

I think that could be fixed in a specific commit after check how it is used on old build script that I don't know so well currently.

Reply:
In that case, that is exactly what was the wanted behavior :)